From eeb491a848b19acd9c84e86a3e7093f37a359d3a Mon Sep 17 00:00:00 2001
From: Vitaly Sokolov <vsokolov@google.com>
Date: Thu, 24 Sep 2026 21:22:28 +0000
Subject: [PATCH] avcodec/opus/silk: zero-initialize active and redundancy
 arrays

In ff_silk_decode_superframe, when coded_channels is 1, redundancy[1]
and active[1] are not read from the bitstream. However, decoding LBRR
frames accesses redundancy[1] when j == 0:
    int active1 = (j == 0 && !(redundancy[1] & (1 << i))) ? 0 : 1;
which causes an uninitialized variable read under MemorySanitizer.

Zero-initialize the active and redundancy arrays to prevent reading
uninitialized values.

Signed-off-by: Vitaly Sokolov <vsokolov@google.com>
Signed-off-by: Marvin Scholz <epirat07@gmail.com>
(cherry picked from commit f3daf49b50edfb60a280a0c1c5bf1438d58c1b5b)
---
 libavcodec/opus/silk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/opus/silk.c b/libavcodec/opus/silk.c
index f7b79fff45d8..9327178e77f2 100644
--- a/libavcodec/opus/silk.c
+++ b/libavcodec/opus/silk.c
@@ -789,7 +789,7 @@ int ff_silk_decode_superframe(SilkContext *s, OpusRangeCoder *rc,
                               int coded_channels,
                               int duration_ms)
 {
-    int active[2][6], redundancy[2];
+    int active[2][6] = {{0}}, redundancy[2] = {0};
     int nb_frames, i, j;
 
     if (bandwidth > OPUS_BANDWIDTH_WIDEBAND ||
